### PR TITLE
Fast follows on 0.34

### PR DESCRIPTION
**Pull request by charlesBochet**

### CODE REVIEW

File: packages/twenty-docker/docker-compose.yml [hunk 1]
```diff
@@ -24,7 +24,9 @@ services:
       PORT: 3000
       PG_DATABASE_URL: postgres://${PGUSER_SUPERUSER:-postgres}:${PGPASSWORD_SUPERUSER:-postgres}@${PG_DATABASE_HOST:-db:5432}/default
       SERVER_URL: ${SERVER_URL}
-      FRONT_BASE_URL: ${FRONT_BASE_URL:-$SERVER_URL}
+      FRONT_DOMAIN: ${FRONT_DOMAIN}
```
Comment (charlesBochet) [inline packages/twenty-docker/docker-compose.yml:27]: not needed

File: packages/twenty-docker/docker-compose.yml [hunk 2]
```diff
@@ -54,9 +56,10 @@ services:
     environment:
       PG_DATABASE_URL: postgres://${PGUSER_SUPERUSER:-postgres}:${PGPASSWORD_SUPERUSER:-postgres}@${PG_DATABASE_HOST:-db:5432}/default
       SERVER_URL: ${SERVER_URL}
-      FRONT_BASE_URL: ${FRONT_BASE_URL:-$SERVER_URL}
+      FRONT_DOMAIN: ${FRONT_DOMAIN}
```
Comment (charlesBochet) [inline packages/twenty-docker/docker-compose.yml:59]: same

File: packages/twenty-docker/k8s/manifests/deployment-server.yaml [hunk 3]
```diff
@@ -37,8 +37,12 @@ spec:
               value: 3000
             - name: SERVER_URL
               value: "https://crm.example.com:443"
-            - name: FRONT_BASE_URL
-              value: "https://crm.example.com:443"
+            - name: FRONT_DOMAIN
```
Comment (charlesBochet) [inline packages/twenty-docker/k8s/manifests/deployment-server.yaml:40]: remove for now

File: packages/twenty-server/src/engine/core-modules/domain-manager/service/domain-manager.service.ts [hunk 4]
```diff
@@ -21,10 +21,26 @@ export class DomainManagerService {
...
+
+    if (
+      !this.environmentService.get('FRONT_PROTOCOL') ||
+      !this.environmentService.get('FRONT_DOMAIN') ||
+      !this.environmentService.get('FRONT_PORT')
+    ) {
+      baseUrl = new URL(this.environmentService.get('SERVER_URL'));
+    } else {
+      baseUrl = new URL(
+        `${this.environmentService.get('FRONT_PROTOCOL')}://${this.environmentService.get('FRONT_DOMAIN')}:${this.environmentService.get('FRONT_PORT')}`,
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/domain-manager/service/domain-manager.service.ts:35]: keep port optional

File: packages/twenty-server/src/engine/core-modules/domain-manager/service/domain-manager.service.ts [hunk 5]
```diff
@@ -21,10 +21,26 @@ export class DomainManagerService {
...
-    const baseUrl = new URL(
-      `${this.environmentService.get('FRONT_PROTOCOL')}://${this.environmentService.get('FRONT_DOMAIN')}`,
-    );
+  getFrontUrl() {
+    let baseUrl: URL;
+
+    if (
+      !this.environmentService.get('FRONT_PROTOCOL') ||
+      !this.environmentService.get('FRONT_DOMAIN') ||
+      !this.environmentService.get('FRONT_PORT')
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/domain-manager/service/domain-manager.service.ts:30]: make it an optional

File: packages/twenty-website/src/content/developers/self-hosting/upgrade-guide.mdx [hunk 6]
```diff
@@ -32,7 +32,12 @@ yarn command:prod upgrade-0.34
 The `yarn database:migrate:prod` command will apply the migrations to the database structure (core and metadata schemas)
 The `yarn command:prod upgrade-0.34` takes care of the data migration of all workspaces.
 
+**Environment Variables**
+
+We have updated the way we handle the frontend URL. If FRONT_DOMAIN or FRONT_PORT are not set, the frontend URL will be set to the server URL.
```
Comment (charlesBochet) [inline packages/twenty-website/src/content/developers/self-hosting/upgrade-guide.mdx:37]: not clear

**Review by Weiko — APPROVED** [review]

File: packages/twenty-server/src/engine/core-modules/domain-manager/service/domain-manager.service.ts [hunk 7]
```diff
@@ -21,10 +21,28 @@ export class DomainManagerService {
...
+
+      const port = this.environmentService.get('FRONT_PORT');
+
+      if (port) {
+        baseUrl.port = port.toString();
+      }
+    }
+
+    return baseUrl;
+  }
```
Comment (AMoreaux) [inline packages/twenty-server/src/engine/core-modules/domain-manager/service/domain-manager.service.ts:42]: The PORT of the SERVER_URL should differ from the frontend.

We should allow to set the FRONT_PORT without the FRONT_PROTOCOL and FRONT_DOMAIN.

Like that:

```suggestion
  getFrontUrl() {
    let baseUrl: URL;

    if (!this.environmentService.get('FRONT_DOMAIN')) {
      baseUrl = new URL(this.environmentService.get('SERVER_URL'));
    } else {
      baseUrl = new URL(
        `${this.environmentService.get('FRONT_PROTOCOL')}://${this.environmentService.get('FRONT_DOMAIN')}`,
      );

      
    }
const port = this.environmentService.get('FRONT_PORT');

      if (port) {
        baseUrl.port = port.toString();
      }
    return baseUrl;
  }
```

Comment (charlesBochet) [reply packages/twenty-server/src/engine/core-modules/domain-manager/service/domain-manager.service.ts:42]: Our main concern is to lower the number of env variables needed
- in self hosting main case: we want front to be served on same url as server and we only have to specify SERVER_URL
- in local development case: we only have to specify FRONT_DOMAIN and FRONT_PORT in .env

Actually, I don't disagree with you, we could change port and protocol in separate if conditions and achieve the same goal

@Weiko what do you think?

**Review by DillonMercz — COMMENTED** [review]
good enough

**Pull request merged**